wandb/apis/importers/base.py

```diff
@@ -0,0 +1,309 @@
+import json
+import platform
+from abc import ABC, abstractmethod
+from concurrent.futures import ProcessPoolExecutor, as_completed
+from contextlib import contextmanager
+from typing import Any, Dict, Iterable, List, Optional, Tuple
+
+from tqdm.auto import tqdm
+
+import wandb
+from wandb.proto import wandb_internal_pb2 as pb
+from wandb.proto import wandb_telemetry_pb2 as telem_pb
+from wandb.sdk.interface.interface import file_policy_to_enum
+from wandb.sdk.interface.interface_queue import InterfaceQueue
+from wandb.sdk.internal.sender import SendManager
+
+Name = str
+Path = str
+
+
+def coalesce(*arg: Any) -> Any:
+    """Return the first non-none value in the list of arguments.  Similar to ?? in C#"""
+    return next((a for a in arg if a is not None), None)
+
+
+@contextmanager
+def send_manager(root_dir):
+    sm = SendManager.setup(root_dir, resume=False)
+    try:
+        yield sm
+    finally:
+        # flush any remaining records
+        while sm:
+            data = next(sm)
+            sm.send(data)
+        sm.finish()
+
+
+class Run:
```

Comment:
Consider calling this ImporterRun or something to differentiate between the 2 other Run's we have
Maybe even _ImporterRun